deps/rabbitmq_mqtt/src/rabbit_mqtt_processor.erl

```diff
@@ -5,651 +5,855 @@
+               host :: inet:ip_address(),
+               port :: inet:port_number(),
+               peer_host :: inet:ip_address(),
+               peer_port :: inet:port_number(),
+               connected_at :: pos_integer()}).
+
+-record(state,
+        {socket,
+         proto_ver :: option(mqtt310 | mqtt311),
+         queue_states = rabbit_queue_type:init() :: rabbit_queue_type:state(),
+         %% Packet IDs published to queues but not yet confirmed.
+         unacked_client_pubs = rabbit_mqtt_confirms:init() :: rabbit_mqtt_confirms:state(),
+         %% Packet IDs published to MQTT subscribers but not yet acknowledged.
+         unacked_server_pubs = #{} :: #{packet_id() => QueueMsgId :: non_neg_integer()},
+         %% Packet ID of next PUBLISH packet (with QoS > 0) sent from server to client.
+         %% (Not to be confused with packet IDs sent from client to server which can be the
+         %% same IDs because client and server assign IDs independently of each other.)
+         packet_id = 1 :: packet_id(),
+         client_id :: option(binary()),
+         clean_sess :: option(boolean()),
+         will_msg :: option(mqtt_msg()),
+         exchange :: option(rabbit_exchange:name()),
+         subscriptions = #{} :: #{Topic :: binary() => QoS :: ?QOS_0..?QOS_1},
+         %% Set if client has at least one subscription with QoS 1.
+         queue_qos1 :: option(rabbit_amqqueue:name()),
+         %% Did the client ever sent us a PUBLISH packet?
+         published = false :: boolean(),
+         ssl_login_name :: none | binary(),
+         retainer_pid :: option(pid()),
+         auth_state :: option(#auth_state{}),
+         peer_addr :: inet:ip_address(),
+         send_fun :: fun((Packet :: tuple(), state()) -> term()),
+         register_state,
+         conn_name :: option(binary()),
+         info :: option(#info{}),
+         delivery_flow :: flow | noflow,
+         %% quorum queues and streams whose soft limit has been exceeded
+         queues_soft_limit_exceeded = sets:new([{version, 2}]) :: sets:set(),
+         qos0_messages_dropped = 0 :: non_neg_integer()
+        }).
+
+-opaque state() :: #state{}.
+
+-spec initial_state(Socket :: any(), ConnectionName :: binary()) ->
+    state().
+initial_state(Socket, ConnectionName) ->
+    {ok, {PeerAddr, _PeerPort}} = rabbit_net:peername(Socket),
+    initial_state(Socket,
+                  ConnectionName,
+                  fun serialise_and_send_to_client/2,
+                  PeerAddr).
+
+-spec initial_state(Socket :: any(),
+                    ConnectionName :: binary(),
+                    SendFun :: fun((mqtt_packet(), state()) -> any()),
+                    PeerAddr :: inet:ip_address()) ->
+    state().
+initial_state(Socket, ConnectionName, SendFun, PeerAddr) ->
+    Flow = case rabbit_misc:get_env(rabbit, mirroring_flow_control, true) of
+               true   -> flow;
+               false  -> noflow
+           end,
+    #state{socket         = Socket,
```

Comment:
I think it would be lovely when making this many changes to a module to try to move away from the hard to read redundant whitespace formatting style.... :)

Reply:
Agreed :) I will run a formatter on the entire `rabbitmq_mqtt` and `rabbitmq_web_mqtt` plugins because I don't want to manually move whitespaces around :)
I will do so in a separate PR because this PR becomes hard to diff and hard to rebase otherwise.